text/0000-prelude-2021-traits.md

```diff
@@ -0,0 +1,125 @@
+- Feature Name: `prelude_2021_traits`
+- Start Date: 2021-02-16
+- RFC PR: [rust-lang/rfcs#0000](https://github.com/rust-lang/rfcs/pull/0000)
+- Rust Issue: [rust-lang/rust#0000](https://github.com/rust-lang/rust/issues/0000)
+
+# Summary
+[summary]: #summary
+
+A new prelude for the 2021 edition, featuring several extra traits.
+
+# Motivation
+[motivation]: #motivation
+
+While types and free functions can be added to the prelude independent of edition boundaries, the same is not true for traits. Adding a trait to the prelude can cause compatibility issues because calls to methods named the same as methods of the newly in scope traits can become ambiguous. While such changes are technically
+[minor][minor-change], such that some backwards compatibility breaks are potentially allowed, the possibility of relatively widespread breakage makes it less attractive to do so without an edition opt-in.
+
+The expected outcome is to make the newly added traits more accessible to users, potentially stimulating their use compared to alternatives.
+
+[minor-change]: https://rust-lang.github.io/rfcs/1105-api-evolution.html#minor-change-adding-a-defaulted-item
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The compiler currently brings all items from `std::prelude::v1` into scope for each module (for code using `std`). Crates that declare usage of the 2021 edition will instead get items from `std::prelude::edition2021` imported into scope. With the implementation of this RFC, we will add `edition2015` and `edition2018` modules in `std::prelude`, and a crate configuring a particular edition will get the prelude appropriate for its edition. The `v1` module will stick around for compatibility reasons, although it might be deprecated in the future.
```

Comment:
In diagnostics and documentation, we usually say "Rust 2021" instead of "edition 2021". So in https://github.com/rust-lang/rust/pull/82217 I called these `std::prelude::rust_2021` (etc.)

Reply:
Updated these.